### PR TITLE
Update existing SSH permission requests to match new backend data structures

### PR DESCRIPTION
**Pull request by GGonryun**
This PR changes the permission request data structures for AWS, Azure, and Google Cloud SSH to be in line with our most recent changes to the P0 backend.

These change impact the construction of SSH commands. This latest version is necessary to continue making valid SSH requests.

### CODE REVIEW

File: src/plugins/google/types.ts [hunk 1]
```diff
@@ -12,14 +12,13 @@ import { PermissionSpec } from "../../types/request";
 import { CliPermissionSpec } from "../../types/ssh";
 import { CommonSshPermissionSpec } from "../ssh/types";
 
-export type GcpSshPermission = {
-  spec: CommonSshPermissionSpec & {
+export type GcpSshPermission = CommonSshPermissionSpec & {
+  zone: string;
+  provider: "gcloud";
+  resource: {
```
Comment (p0-andrewa) [inline src/plugins/google/types.ts:18]: Nit: Should `provider` be the first field here? That seems to be the pattern I've observed with other discriminated types.

**Review by p0-andrewa — COMMENTED** [review]
LGTM, just a small style nit/question.

**Review by p0-andrewa — APPROVED** [review]
Oops, clicked too quickly and forgot to actually select "Approve".

File: src/plugins/aws/ssh.ts [hunk 2]
```diff
@@ -112,11 +112,12 @@ export const awsSshProvider: SshProvider<
...
-      permission.spec;
-    const { idcId, idcRegion } = awsResourcePermission.permission;
-    const { ssh, name } = generated;
-    const { linuxUserName } = ssh;
+    const { resource, region } = permission;
+    const { idcId, idcRegion, instanceId, accountId } = resource;
+    const {
+      linuxUserName,
+      resource: { name },
+    } = generated;
```
Comment (nbrahms) [inline src/plugins/aws/ssh.ts:120]: Nested destructuring gets a bit verbose and a bit complicated:

```suggestion
    const { linuxUserName, resource } = generated;
    const { name } = resource;
```

**Review by nbrahms — APPROVED** [review]
I'd probably re-write the PR description. I believe this change is more about updating for the new permission-request data structure used by the P0 backend.

Other than that, just the tiniest of style feedback.

File: src/plugins/aws/ssh.ts (hunk 2 above)
Comment (GGonryun) [reply src/plugins/aws/ssh.ts:120]: There's a collision between `permission.resource` and `generated.resource` but i used an alias to make it work 👍

**Pull request merged**